### PR TITLE
DTSPO-2173  Mirgrate JD public

### PR DESCRIPTION
**Pull request by mokainos**
https://tools.hmcts.net/jira/browse/DTSPO-2173

### CODE REVIEW

File: k8s/environments/stg/common/juror-digital/jd-public.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+apiVersion: flux.weave.works/v1beta1
+kind: HelmRelease
+metadata:
+  name: jd-public
+  namespace: jd
+  annotations:
+    flux.weave.works/automated: "false"
```
Comment (timja) [inline k8s/environments/stg/common/juror-digital/jd-public.yaml:8]: use fluxcd annotations not the old ones

File: k8s/environments/stg/common/juror-digital/jd-public.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    git: git@github.com:hmcts/sds-helm-charts
+    path: charts/jd-public
+    ref: master
+  values:
+    nodejs:
+      image: ssprivatestg.azurecr.io/jd/juror_digital_public_portal:2.21.427
+      ingressHost: 'jdfrontend-nle.platform.hmcts.net'
+      ingressSessionAffinity:
+        enabled: true
+        sessionCookieName: ""
```
Comment (timja) [inline k8s/environments/stg/common/juror-digital/jd-public.yaml:22]: why does this need to be set like this? we can fix the chart upstream if required

File: k8s/environments/stg/common/juror-digital/jd-public.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      environment:
```
Comment (timja) [inline k8s/environments/stg/common/juror-digital/jd-public.yaml:23]: should be in the chart

**Review by timja — CHANGES_REQUESTED** [review]
this should be defined with kustomize, using a base and then a patch to override the image and ingress per environment

File: k8s/environments/stg/common-overlay/juror-digital/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (timja) [inline k8s/environments/stg/common-overlay/juror-digital/kustomization.yaml:2]: don't you need to add this kustomiation to the stg common overlay as well?

Comment (mokainos) [reply k8s/environments/stg/common-overlay/juror-digital/kustomization.yaml:2]: modified it

**Review by timja — APPROVED** [review]

**Pull request merged**